File: pascals.py
# Time Complexity : O(n^2)
# Space Complexity :O(n)
# Did this code successfully run on Leetcode : yes
# Any problem you faced while coding this : No

# create a list of list -number of rowa equal to numrows inputed and number of lements in each row will be from 0 to the row id except for first row
# create a temp array and add it to the resulting list of list

class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        res=[]
        res.append([1])
        for i in range(1,numRows):
            temp=[]
            temp.append(1)
            for j in range(1,i):
                print("inside")
                temp.append(res[i-1][j-1]+res[i-1][j])
            temp.append(1)
            res.append(temp)       
        return res
        

File: Pascals.java
//TC = O(rows*columns)
//SC = O(1)
//// Did this code successfully run on Leetcode :yes
class Pascals{
    public List<List<Integer>> generate(int rows){
        List<List<Integer>> res = new ArrayList<>();
        for(int i=0; i<rows; i++){
            List<Integer> list = new ArrayList<>();
            for(int j = 0; j<=i; j++){ // 
                if(j==0 || j==i) list.add(1); // As first and last element in each rpw will always be 1.
                else{
                    int left = res.get(i-1).get(j-1);  // Going previous row and elemnt on left.
                    int right = res.get(i-1).get(j); // Going previous row and element on rght.
                    list.add(left+right);
                }
            }
            res.add(list);
        }
        return res;
    }
}